ur_driver/test_move.py

```diff
@@ -1,106 +1,156 @@
-    g = FollowJointTrajectoryGoal()
-    g.trajectory = JointTrajectory()
-    g.trajectory.joint_names = JOINT_NAMES
-    g.trajectory.points = [
-        JointTrajectoryPoint(positions=Q1, velocities=[0]*6, time_from_start=rospy.Duration(2.0)),
-        JointTrajectoryPoint(positions=Q2, velocities=[0]*6, time_from_start=rospy.Duration(3.0)),
-        JointTrajectoryPoint(positions=Q3, velocities=[0]*6, time_from_start=rospy.Duration(4.0))]
+#Resends the same goal after 3 seconds delay
+#The already running goal is cancelled and the arm moves back to the start pose before executing the trajectory again
+def move_interrupt(_traj_ac, _jnt_names):
+  q1, q2, q3 = get_traj_pts()
+  g = get_trajectory_goal(_jnt_names, q1, q2, q3)
+  _traj_ac.send_goal(g)
+  time.sleep(3.0)
+  print("Interrupting")
+  handle_send_goal(_traj_ac, g)
+
+def get_joint_state():
+  return rospy.wait_for_message('/joint_states', JointState)
+
+def get_current_joint_vals():
+  return list(get_joint_state().position)
+
+def get_joint_names():
+  return get_joint_state().name
+  
+#Adds/subtracts 0.3 radians to the last 2 joint values
+def get_traj_pts():
+  pos = get_current_joint_vals()
+  q1 = pos[:4] + [x + 0.3 for x in pos[4:]]
+  q2 = pos[:4] + [x - 0.3 for x in pos[4:]]
+  q3 = pos
+  return q1, q2, q3
+
+def get_controller():
+  _list_cm = ControllerManagerLister()
+  controller_list = ControllerLister(_list_cm()[0])
+  _list_controllers = controller_list()
     
-    client.send_goal(g)
-    time.sleep(2.0)
-    print "Interrupting"
-    client.send_goal(g)
-    try:
-        client.wait_for_result()
-    except KeyboardInterrupt:
-        client.cancel_goal()
-        raise
+  jtc_list = filter_by_type(_list_controllers,
+                            'JointTrajectoryController',
+                            match_substring=True)
+  running_jtc_list = filter_by_state(jtc_list, 'running')
+  return running_jtc_list[0]
 
+def user_menu():
+  choice=True
+  print("""
+  1.Perform a pre-defined sequence of joint movements once
+  2.Perform a pre-defined sequence of joint movements in a loop
+  3.Set joint values in JointTrajectory in a disordered way
+    and perform a pre-defined sequence of joint movements once
+  4.Send a pre-defined joint trajectory twice 
+    cancels the first goal and starts the new one
+  5.Exit/Quit
+  """)
+  choice=raw_input("What would you like to do? ") 
+  return choice
+    
 def main():
-    global client
-    try:
-        rospy.init_node("test_move", anonymous=True, disable_signals=True)
-        client = actionlib.SimpleActionClient('follow_joint_trajectory', FollowJointTrajectoryAction)
-        print "Waiting for server..."
-        client.wait_for_server()
-        print "Connected to server"
-        #move1()
-        move_repeated()
-        #move_disordered()
-        #move_interrupt()
-    except KeyboardInterrupt:
-        rospy.signal_shutdown("KeyboardInterrupt")
-        raise
+  _use_ros_control = False
```

Comment:
@ipa-hsd `_use_ros_control`?? what does it mean?

Reply:
I wanted to read the controller's namespace in case of ros_control. Else the user passes the namespace as an arugument.

Reply:
removed this to avoid confusion